### PR TITLE
add idle_timeout to fix connection was reset by the upstream where keep-alive is open

### PR DESCRIPTION
**Pull request by devincd**
Bug fix [envoyproxy](https://github.com/envoyproxy/envoy/issues/1979)

### CODE REVIEW

**Comment by Devin-caodi** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here (e.g. `I signed it!`) and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google.com/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).

I signed it!

**Comment by sebastienvas** [comment]
Just verified that it was ok to run on our infra, assigning from someone in the network wg.

/assign @costinm

**Comment by Devin-caodi** [comment]
@costinm What's going on with this PR?

**Comment by Devin-caodi** [comment]
@sebastienvas What's going on with this PR?

**Comment by devincd** [comment]
ping

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 1]
```diff
@@ -323,6 +327,29 @@ func applyTrafficPolicy(cluster *v2.Cluster, policy *networking.TrafficPolicy, p
 	applyUpstreamTLSSettings(cluster, tls)
 }
 
+func applyCommonHttpProtocolOptions(cluster *v2.Cluster) {
+	idletimeout := intEnv("IDLE_TIMEOUT", 0)
```
Comment (rshriram) [inline pilot/pkg/networking/core/v1alpha3/cluster.go]: This should go as a global mesh config option in api/mesh/config.proto. You should then read it here and apply to the config. 
See how connect timeout is implemented.

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 2]
```diff
@@ -29,6 +31,7 @@ import (
 	"istio.io/istio/pilot/pkg/model"
 	"istio.io/istio/pilot/pkg/networking/util"
 	"istio.io/istio/pkg/log"
+	envoy_api_v2_core5 "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
```
Comment (rshriram) [inline pilot/pkg/networking/core/v1alpha3/cluster.go]: can you call this as envoycore or something?

Comment (devincd) [reply pilot/pkg/networking/core/v1alpha3/cluster.go]: yes

**Pull request closed without merging**